libr/bin/p/bin_xtr_fatmach0.c

```diff
@@ -104,17 +125,33 @@ static RBinXtrData * oneshot(RBin *bin, const ut8 *buf, ut64 size, int idx) {
 	struct r_bin_fatmach0_arch_t *arch;
 	RBinXtrData *res = NULL;
 	int narch;
-
+	struct MACH0_(mach_header) *hdr;
```

Comment:
Here in oneshot apply the same logic as we did in dyldcache. Is not needed to create xtr_obj each time we call oneshot. Extend the load function to follow the same pattern as dyldcache and use bin->cur->xtr_obj instead